src/utils/read_all.c

```diff
@@ -37,8 +46,13 @@ ssize_t write_all (int fd, const void *buf, size_t len)
     return count;
 }
 
-ssize_t read_all (int fd, void **bufp)
+ssize_t read_all (const char* filename, void **bufp)
```

Comment:
This interface probably should take a page aligned memory address as an input for the UCX managed memory. So, the allocation would happen outside of this function.

Reply:
That is the buffer bufp. Once we do the UCX, we can move allocation there.